overlord/snapstate/dbus_test.go

```diff
@@ -0,0 +1,344 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2020 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package snapstate_test
+
+import (
+	"context"
+
+	. "gopkg.in/check.v1"
+
+	"github.com/snapcore/snapd/overlord/configstate/config"
+	"github.com/snapcore/snapd/overlord/snapstate"
+	"github.com/snapcore/snapd/snap"
+)
+
+func (s *snapmgrTestSuite) TestCheckDBusServiceConflictsSystem(c *C) {
+	yamlFragment := `slots:
+  dbus-slot:
+    interface: dbus
+    bus: system
+    name: org.example.Foo
+apps:
+  daemon:
+    daemon: simple
+    activates-on: [dbus-slot]
+`
+
+	someSnap, err := snap.InfoFromSnapYaml([]byte("name: some-snap\n" + yamlFragment))
+	c.Assert(err, IsNil)
+	otherSnap, err := snap.InfoFromSnapYaml([]byte("name: other-snap\n" + yamlFragment))
+	c.Assert(err, IsNil)
+
+	restore := snapstate.MockSnapReadInfo(func(name string, si *snap.SideInfo) (*snap.Info, error) {
+		switch name {
+		case "some-snap":
+			return someSnap, nil
+		case "other-snap":
+			return otherSnap, nil
+		default:
+			return s.fakeBackend.ReadInfo(name, si)
+		}
+	})
+	defer restore()
+
+	s.state.Lock()
+	defer s.state.Unlock()
+	si := &snap.SideInfo{
+		RealName: "other-snap",
+		Revision: snap.R(-42),
+	}
+	snapstate.Set(s.state, "other-snap", &snapstate.SnapState{
+		Active:   true,
+		Sequence: []*snap.SideInfo{si},
+		Current:  si.Revision,
+		SnapType: "app",
+	})
+
+	err = snapstate.CheckDBusServiceConflicts(s.state, someSnap)
+	c.Assert(err, ErrorMatches, `snap "some-snap" requesting to activate on system bus name "org.example.Foo" conflicts with snap "other-snap" use`)
+}
+
+func (s *snapmgrTestSuite) TestCheckDBusServiceConflictsSession(c *C) {
+	yamlFragment := `slots:
```

Comment:
Good idea.  That cut almost 50 lines from the tests.  Rather than having yaml fragments as constants, I decided to make them format strings I can pass to `fmt.Sprintf()` to set the snap name.  That was the only thing any of the tests added to their fragments.